### PR TITLE
Upgrading to Neutrino v5

### LINKED ISSUE #4: Warn that `babel-runtime` will be embedded if not declared as dependency
I think we should require that `babel-runtime` is declared a dependency... Right now if it's not present in `package.json` it'll be embedded in `index.bundle.js`, that seems like a really bad thing.

@eliperelman, in a node project if we have two libraries requiring `babel-runtime: "^6.0.0"` then I think they'll both get the same instance of babel-runtime, hence, it won't be loaded into memory twice.

### LINKED ISSUE #5: "source-map-support" is always installed
Is this intentional?

If so we should definitely require that it's declared a dependency... But honestly, I'm not sure libraries should be shipping with source-map-support embedded...

I suspect that there is a few things we haven't thought through with respect to using neutrino for modules. This choice seems reasonably sane for services... Granted it should still be a requirement.

### PR DESCRIPTION
**Pull request by eliperelman**
Upgrading to Neutrino v5.

Fixes #5: source-map-support via banner is automatic, only if the application has it installed in package.json.

Partially addresses #6: using `import()`, code can be conditionally and dynamically imported.

Also fixes #4 since external dependencies are not included in bundles.

### CODE REVIEW

**Comment by eliperelman** [comment]
Another thing, with the latest Node.js preset, testing isn't baked in. It's easy enough to just add `neutrino-preset-mocha` to a project though to get this functionality back on any project.

**Comment by eliperelman** [comment]
@imbstack @jonasfj if either of you would like to give a review here too, I'd appreciate it! 😃

File: README.md [hunk 1]
```diff
@@ -1,89 +1,361 @@
...
-    "test": "neutrino test"
-  }
+  ]
 }
 ```
 
-Add your source code to `src/`, which is compiled using Babel's es2015 preset.
+_Note: While this preset works well for many types of Node.js applications, it's important to make the distinction
+between applications and libraries. This preset will not work optimally out of the box for creating distributable
+libraries, and will take a little extra customization to make them suitable for that purpose._
```
Comment (djmitche) [inline README.md:157]: I assume this is numerically most uses of this preset, since we really only have one application?

**Review by djmitche — APPROVED** [review]
This looks fine to me, but I'm not sure what I would find objectionable :)

**Review by imbstack — APPROVED** [review]
Seems good to me!

File: README.md (hunk 1 above)
Comment (eliperelman) [reply README.md:157]: Actually, I don't think this preset is used anywhere yet. This preset is for Node.js apps, i.e. can be used on services, etc.

Comment (eliperelman) [reply README.md:157]: The web stuff uses different presets.

**Comment by djmitche** [comment]
Oh, I see! In that case, are there any bundles to have their trees shaken?  And how does hot module reloading from source work in an active application?  I guess I'd have to see it in use to really understand..

**Comment by eliperelman** [comment]
I guess tree shaking isn't super-compelling for external bundles obviously, but still serves a purpose for local code. Hot module replacement is more of a development feature, allowing you to make changes and have the process/module replaced.

Little demo of a Node.js server that logs the result of an imported function (`a + b = 13`), then while the server is running, the module switches to multiplication (`a * b = 42`).

node_hmr

**Comment by djmitche** [comment]
OK, well, I'm certainly happy to see this land.  I'd want to play around with it with a service before we merge it there.  I find tools pretty hard to hack on since the build process takes so long, and I don't want to have the same experience with other services.

**Comment by eliperelman** [comment]
Play away, and please file issues if something doesn't seem right. :)

**Comment by eliperelman** [comment]
@djmitche the latest Neutrino should help speed things up a bit. Using HMR for Node.js will also make development faster since you wouldn't necessarily need to kill and restart the process for every change you make.

**Pull request merged**